docs/Using-Fleet/configuration-files/README.md

```diff
@@ -264,15 +270,928 @@ spec:
+  ```
+  webhook_settings:
+    failing_policies_webhook:
+      enable_failing_policies_webhook: true
+  ```
+
+##### webhook_settings.failing_policies_webhook.destination_url
+
+The URL to `POST` to when the condition for the webhook triggers.
+
+- Optional setting, required if webhook is enabled (string).
+- Default value: none.
+- Config file format:
+  ```
+  webhook_settings:
+    failing_policies_webhook:
+      destination_url: "https://example.org/webhook_handler"
+  ```
+
+##### webhook_settings.failing_policies_webhook.policy_ids
+
+The IDs of the policies for which the webhook will be enabled.
+
+- Optional setting (array of integers).
+- Default value: empty.
+- Config file format:
+  ```
+  webhook_settings:
+    failing_policies_webhook:
+      policy_ids:
+        - 1
+        - 2
+        - 3
+  ```
+
+##### webhook_settings.failing_policies_webhook.host_batch_size
+
+Maximum number of hosts to batch on `POST` requests. A value of `0`, the default, means no batching, all hosts failing a policy will be sent on one `POST` request.
+
+- Optional setting (integer).
+- Default value: `0`.
+- Config file format:
+  ```
+  webhook_settings:
+    failing_policies_webhook:
+      host_batch_size: 100
+  ```
+
+#### Recent vulnerabilities
+
+The following options allow the configuration of a webhook that will be triggered if recently published vulnerabilities are detected and there are affected hosts. A vulnerability is considered recent if it has been published in the last 2 days (based on the National Vulnerability Database, NVD).
+
+Note that the recent vulnerabilities webhook is not checked at `webhook_settings.interval` like other webhooks - it is checked as part of the vulnerability processing and runs at the `vulnerabilities.periodicity` interval specified in the [fleet configuration](../../Deploying/Configuration.md#periodicity).
+
+##### webhook_settings.vulnerabilities_webhook.enable_vulnerabilities_webhook
+
+Defines whether to enable the vulnerabilities webhook.
+
+- Optional setting (boolean).
+- Default value: `false`.
+- Config file format:
+  ```
+  webhook_settings:
+    vulnerabilities_webhook:
+      enable_vulnerabilities_webhook: true
+  ```
+
+##### webhook_settings.vulnerabilities_webhook.destination_url
+
+The URL to `POST` to when the condition for the webhook triggers.
+
+- Optional setting, required if webhook is enabled (string).
+- Default value: none.
+- Config file format:
+  ```
+  webhook_settings:
+    vulnerabilities_webhook:
+      destination_url: "https://example.org/webhook_handler"
+  ```
+
+##### webhook_settings.vulnerabilities_webhook.host_batch_size
+
+Maximum number of hosts to batch on `POST` requests. A value of `0`, the default, means no batching, all hosts affected will be sent on one `POST` request.
+
+- Optional setting (integer).
+- Default value: `0`.
+- Config file format:
+  ```
+  webhook_settings:
+    vulnerabilities_webhook:
+      host_batch_size: 100
+  ```
 
 ### Agent options
```

Comment:
Agent options documentation would largely remain unchanged, as this section already links to the osquery documentation for all available fields (I assume we still want to do that) and some more complex config options are well explained (such as overrides with auto table constructions, and YARA configuration).

Reply:
>  links to the osquery documentation for all available fields

At some point, it would be great to document these fields in the Fleet docs. I think we can come back to this in a later pass.

Validation will still apply to all the agent options fields, correct?

Reply:
> Validation will still apply to all the agent options fields

Yes, that's correct.